### ISSUE TITLE
Contacts API fails with Content-Type application/json; charset=utf-8

### ISSUE THREAD

**Issue by dustinmoris** [sendgrid/sendgrid-csharp#910]
Making a request to the Contacts API endpoint (`{baseUrl}/marketing/contacts`) with a Content-Type of `application/json; charset=utf-8` returns a 400 Bad Request saying that the content type is not supported:

```
{
    "errors": [
        {
            "field": "content-type",
            "message": "invalid content-type: application/json; charset=utf-8"
        }
    ]
}
```

You can try it out yourself at https://sendgrid.com/docs/API_Reference/api_v3.html and send a PUT request to the Contacts endpoint.

Unfortunately this C# client ALWAYS sets the charset to UTF8, which makes it unusable for some endpoints like the one above. Can the SendGrid folks please accept the entirely valid media type with the charset or at least make it optional in the RequestAsync method of the client?

Feels really broken right now and is frustrating that I've spend a couple hours trying to understand why a valid request gets denied for appending the charset -.-

If I send the same request with just `application/json` (without the utf-8 charset) then the request succeeds, but that has to be done manually and can't be done with the RequestAsync method at the moment.

**Comment by thinkingserious** [comment]
Thanks for reporting this @dustinmoris,

We've added this to our backlog for further investigation.

With best regards,

Elmer

**Comment by shaunroot** [comment]
Is there a workaround for this using the SendGrid client or should I just drop it and use the standard RestClient?

**Comment by dustinmoris** [comment]
@shaunroot I ended up just using the plain `HttpClient` in .NET Core for now, but looking at the code I believe you could just use [this constructor](https://github.com/sendgrid/sendgrid-csharp/blob/master/src/SendGrid/SendGridClient.cs#L100) and pass in your own request headers with your own `ContentType`, which I think will override the default Content-Type.

I haven't tested this though, just my feeling by briefly looking at the code!

**Comment by hallidev** [comment]
I can't believe this is an issue nearly four months after being reported.

For a hack, add this on line 231 of SendGridClient.cs:

`                if (request.Content != null)
                {
                    request.Content.Headers.ContentType.CharSet = string.Empty;
                }`

I have no idea if this breaks any other API calls.

This needs to be fixed on your server. It doesn't seem like an earth shattering change. Why has it not been addressed? The fact that API calls fail with your own client is unacceptable.

**Comment by marius-stanescu** [comment]
Any updates on this?

**Comment by fpavlic** [comment]
Any updates?

**Comment by childish-sambino** [comment]
Closing this as the consensus is that this is a server-side issue that the client shouldn't need a workaround/hack to resolve. Recommend contacting [support](http://support.sendgrid.com) to investigate a back-end fix.

**Comment by dustinmoris** [comment]
@childish-sambino Are you guys not working in the same company? Is [https://github.com/sendgrid/sendgrid-csharp](https://github.com/sendgrid/sendgrid-csharp) not part of the company SendGrid? If yes, then consider this issue the support ticket. If it's the wrong repo then please move it into the correct (private?) repo where it belongs?

**Comment by hallidev** [comment]
@childish-sambino I'm wondering the same thing. Isn't this Sendgrid's official client for C#?

**Comment by childish-sambino** [comment]
@dustinmoris Yes, this repo is part of Twilio SendGrid, but issues here are only monitored by the engineering team responsible for managing the repo itself. For account, request, and API-specific issues, only [support](http://support.sendgrid.com/) is able to help.

**Comment by childish-sambino** [comment]
@hallidev Yes, this is the official C# client, but as the issue is with the API itself it is not something that should/can be resolved here.

**Comment by hallidev** [comment]
Then please open up an internal ticket. We're paying customers and shouldn't have to hunt down people to fix this. That support link you gave is literally laughable. I'm honestly offended if you think someone could start with that link and end with a code change to your API.

You have to be kidding me.

**Comment by VagueGit** [comment]
@dustinmoris and @hallidev make a good point. Paying customers should not be told to find the appropriate support person within an organisation. Departments within an organisation should not tell paying customers 'not my problem'. If paying customers raise an issue, getting that issue fixed should not be the customer's problem.

I see this time and again on this repo and it is not a good look for Twilio/SendGrid

**Comment by hallidev** [comment]
Funny enough, your comment and my SendGrid bill just showed up side by side in my email

**Comment by marius-stanescu** [comment]
Thinking of switching to MailChimp ...

**Comment by dustinmoris** [comment]
@childish-sambino Maybe my first response was a bit tongue in cheek. I was thinking since `the consensus is that this is a server-side issue` it doesn't have to be a customer who raises that ticket internally, right? It could be just your team, which maintains the C# client to raise that ticket saying that your client's tests are failing because the server has a bug and then assign it to the server side team? After all you all work towards the same goal I'd be thinking, making SendGrid great, or is every team just looking after their own micro goal, not really being bothered about the overall experience?

**Comment by childish-sambino** [comment]
Ah, able to recreate this locally. Checking with support.

**Comment by childish-sambino** [comment]
Best I can tell is that the default charset for JSON is utf-8, so including this in the content type is not necessary. But, still, it should not be rejected. I'll update the existing PR to check if the media type is the default (json) and, if so, not drop the charset. This will at least unblock usage in this repo. Will follow-up when I have an backend engineering response/ticket.

**Comment by hallidev** [comment]
@childish-sambino Thank you. I apologize for getting harsh yesterday. Appreciate the follow up.

**Comment by tolstenko** [comment]
Sorry to bother you again, I know this is fixed on the CSharp lib. But I am using the bare minimum with axios for js and I am facing the same issue. So the problem is not on the client (axios for js neither the csharp client). Please reconsider moving this to a backend issue.
pls refer to https://datatracker.ietf.org/doc/html/rfc8259 
```
 Note:  No "charset" parameter is defined for this registration.
      Adding one really has no effect on compliant recipients.
```
Correct me if I am wrong but the sendgrid servers should accept charset.
For now, I am going to hardcode this fix, but given others might face the same issue, pls re-open the issue.

Here goes a similar discussion but for axios https://github.com/axios/axios/issues/2154
